### PR TITLE
[pytx] Move Clients to /exchanges/clients

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

The thing you've all been waiting for - the big renames.

Currently we live in this world:

```
- threatexchange
  - ncmec
  - fb_threatexchange
  - stopncii
  - fetcher
    -  apis/
      - FbThreatExchangeSignalExchange.py
      - ... 
    - collab_config.py
    - fetch_api.py
    - fetch_state.py
    - simple/
```
I am moving us to this world:

```
- threatexchange
  - exchanges/
    - clients/
      - ncmec/
      - fb_threatexchange/
      - ...
    - impls/
      - stop_ncii.py  
    - signal_exchange.py
    - fetched_state.py
    - collab_config.py
    - helpers.py  (simple)
```

This diff creates exchanges directory and moves clients

Test Plan
---------

$ py.test
$ mypy

### CODE REVIEW

**Comment by Dcallies** [comment]
I have a bug fix in this stack to help mypy pass (#1054)

File: python-threatexchange/threatexchange/cli/dataset/simple_serialization.py [hunk 1]
```diff
@@ -7,8 +7,10 @@
 import re
 import typing as t
 
-from threatexchange.fb_threatexchange import threat_updates
-from threatexchange.fb_threatexchange.descriptor import SimpleDescriptorRollup
+from threatexchange.exchanges.clients.fb_threatexchange import threat_updates
+from threatexchange.exchanges.clients.fb_threatexchange.descriptor import (
```
Comment (schatten) [inline python-threatexchange/threatexchange/cli/dataset/simple_serialization.py:11]: totally-non-blocking:

Does it smell weird to you that a common serialization class depends on exchanges.clients.fb_threatexchange ?

Is this a candidate for externalizing down the road? Perhaps fb_threatexchange could expose serialization / deserialization methods?

File: python-threatexchange/threatexchange/cli/config_cmd.py [hunk 2]
```diff
@@ -455,7 +455,7 @@ def execute(self, settings: CLISettings) -> None:
         self.action(settings)
 
     def execute_list(self, settings: CLISettings) -> None:
-        signals = settings.get_all_signal_types(default_to_sample=False)
+        signals = settings.get_all_signal_types()
```
Comment (schatten) [inline python-threatexchange/threatexchange/cli/config_cmd.py:458]: With me merging in #1054, this may cause a problem. Apologies.

**Review by schatten — APPROVED** [review]
Per your comment, I expected to see an 'impls' module and was not sure what that was, but I don't see one here.

Happy to merge this in. Definitely a step in the right direction.

File: python-threatexchange/threatexchange/cli/config_cmd.py (hunk 2 above)
Comment (Dcallies) [reply python-threatexchange/threatexchange/cli/config_cmd.py:458]: No, I kept the commits separate, I can use rebase -i to only get my changes, you did the right thing!

File: python-threatexchange/threatexchange/cli/dataset/simple_serialization.py (hunk 1 above)
Comment (Dcallies) [reply python-threatexchange/threatexchange/cli/dataset/simple_serialization.py:11]: Yea, definitely does. Let me trace down what happened.

**Comment by Dcallies** [comment]
impls module is in the followup! Breaking it up over a few commits, and wanted to get your feedback first.

**Pull request merged**